SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialComponentTest/SpatialComponentTest.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "SpatialFunctionalTest.h"
+#include "SpatialComponentTest.generated.h"
+
+class ASpatialComponentTestActor;
+class ASpatialComponentTestReplicatedActor;
+enum class ESpatialHasAuthority : uint8;
+
+/** Check SpatialComponentTest.cpp for Test explanation. */
+UCLASS()
+class SPATIALGDKFUNCTIONALTESTS_API ASpatialComponentTest : public ASpatialFunctionalTest
+{
+	GENERATED_BODY()
+
+public:
+	ASpatialComponentTest();
+
+	virtual void PrepareTest() override;
+
+	virtual void GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const override;
+
+	UFUNCTION(CrossServer, Reliable)
+	void CrossServerSetDynamicReplicatedActor(ASpatialComponentTestReplicatedActor* Actor);
+
+	UPROPERTY(EditAnywhere, Category = "Default")
+	ASpatialComponentTestActor* LevelActor;
+
+	UPROPERTY(EditAnywhere, Category = "Default")
+	ASpatialComponentTestReplicatedActor* LevelReplicatedActor;
+
+	// This needs to be a position that belongs to Server 1.
+	UPROPERTY(EditAnywhere, Category = "Default")
+	FVector Server1Position;
+
+	// This needs to be a position that belongs to Server 2.
+	UPROPERTY(EditAnywhere, Category = "Default")
+	FVector Server2Position;
+
+	UPROPERTY(Replicated)
+	ASpatialComponentTestReplicatedActor* DynamicReplicatedActor;
+
+private:
+	void CheckComponents(ASpatialComponentTestActor* Actor, int ExpectedServerId, int ExpectedClient1ComponentCount = 0,
```

Comment:
I'm not sure these defaults help, might just be easier to be explicit when calling this function

Reply:
Ok. I think it is related to the same issue of having the combined server and client checks together that have been separated in the subsequent task.